paparazzi/src/test/java/app/cash/paparazzi/internal/resources/PseudolocaleGeneratorTest.kt

```diff
@@ -0,0 +1,209 @@
+      }
+    }
+
+    // PLURALS
+    with(result[ResourceNamespace.TODO(), ResourceType.PLURALS]!!) {
+      assertThat(size).isEqualTo(1)
+      with(get("plural_name") as PluralsResourceValue) {
+        assertThat(pluralsCount).isEqualTo(2)
+        assertThat(getValue(0)).isEqualTo("[Ñöţĥîñĝ one two]")
+        assertThat(getValue(1)).isEqualTo("[Öñé Šţŕîñĝ one two]")
+      }
+    }
+
+    // Other resources should not be pseudo localized
+    // BOOL
+    with(result[ResourceNamespace.TODO(), ResourceType.BOOL]!!) {
+      assertThat(size).isEqualTo(2)
+      assertThat(get("screen_small")?.value).isEqualTo(true.toString())
+      assertThat(get("adjust_view_bounds")?.value).isEqualTo(false.toString())
+    }
+
+    // COLOR
+    with(result[ResourceNamespace.TODO(), ResourceType.COLOR]!!) {
+      assertThat(size).isEqualTo(3)
+      assertThat(get("test_color")?.value).isEqualTo("#ffffffff")
+      assertThat(get("test_color_2")?.value).isEqualTo("#00000000")
+      assertThat(get("color_selector")?.value).isEqualTo("$resFolderRoot/color/color_selector.xml")
+    }
+
+    // DIMEN
+    with(result[ResourceNamespace.TODO(), ResourceType.DIMEN]!!) {
+      assertThat(size).isEqualTo(2)
+      assertThat(get("textview_height")?.value).isEqualTo("25dp")
+      assertThat(get("textview_width")?.value).isEqualTo("150dp")
+    }
+  }
+
+  @Test
+  fun testBidiPseudolocalization() {
+    val result = createResourceTable()
+    result.pseudolocalizeIfNeeded(LocaleQualifier.getQualifier("ar-rXB"))
+
+    // STRING
+    with(result[ResourceNamespace.TODO(), ResourceType.STRING]!!) {
+      assertThat(size).isEqualTo(4)
+      assertThat(get("string_name")?.value).isEqualTo("${bidiWordStart}Test$bidiWordEnd ${bidiWordStart}String$bidiWordEnd")
+      assertThat(get("string_name_xliff")?.value).isEqualTo("${bidiWordStart}Test$bidiWordEnd ${bidiWordStart}String$bidiWordEnd $bidiWordStart{0}$bidiWordEnd ${bidiWordStart}with$bidiWordEnd ${bidiWordStart}suffix$bidiWordEnd")
+      assertThat(get("string_name_html")?.value).isEqualTo("${bidiWordStart}This$bidiWordEnd ${bidiWordStart}Test$bidiWordEnd ${bidiWordStart}String$bidiWordEnd")
+      assertThat(get("string_name_html")?.rawXmlValue).isEqualTo("${bidiWordStart}This$bidiWordEnd $bidiWordStart<b>Test</b>$bidiWordEnd ${bidiWordStart}String$bidiWordEnd")
+      assertThat(get("string_not_translate")?.value).isEqualTo("Not Translatable String")
+    }
+
+    // ARRAY
+    with(result[ResourceNamespace.TODO(), ResourceType.ARRAY]!!) {
+      assertThat(size).isEqualTo(1)
+      with(get("string_array_name") as ArrayResourceValue) {
+        assertThat(elementCount).isEqualTo(2)
+        assertThat(getElement(0)).isEqualTo("${bidiWordStart}First$bidiWordEnd ${bidiWordStart}Test$bidiWordEnd ${bidiWordStart}String$bidiWordEnd")
+        assertThat(getElement(1)).isEqualTo("${bidiWordStart}Second$bidiWordEnd ${bidiWordStart}Test$bidiWordEnd ${bidiWordStart}String$bidiWordEnd")
```

Comment:
uber-nit: let's consistently use the curly braces for all the "bidi" val usages; improves readability